### ISSUE TITLE
Parameter override in Assembly Initialize

### ISSUE THREAD

**Issue by JameyRandall** [Magenic/MAQS#14]
We have an API Utility class that is setup to get the authentication cookie that is needed in the test. Currently the urls, parameters and end points are pulled from the app config. I would like to override them from the run settings file. The APIUtility class is getting called as part of the assembly initialize.
 
ex: Config.GetGeneralValue("APICookieURL", "https://default.com")
 
  <MagenicMaqs>
    <add key="APIBaseURI" value ="https://base.com" />
    <add key="APICookieURL" value="https://authlogin.com" />
  </MagenicMaqs>
 
I would like to change these parameters at run time from Azure so we can run against multiple environments (dev, test,  stage... ). To do this I am adding the parameters in the .runsettings file. 
 
ex: Parameter name="MagenicMaqs:APIBaseURI" value ="https://someurl.com" 
    Parameter name="MagenicMaqs:APICookieURL" value="https://otherurl.com"
 
When I run an api test, it is not pulling in the values from the .runsettings file. 

If I move the call to the APIUtility class to the Test Initialize then the parameters are overwritten as expected.

**Comment by TroyWalshProf** [comment]
This works as designed.
That being said, the design was bad.

Temp workaround is to add and call this code where you are doing the assembly initialize 
```
        private void UpdateConfigParameters()
        {
            Dictionary<string, string> passedInParameters = new Dictionary<string, string>();

            try
            {
                if (testContextInstance != null)
                {
                    // Update configuration settings for Visual Studio unit test
                    List<string> propeties = new List<string>();
                    IDictionary<string, object> contextProperties = (IDictionary<string, object>)testContextInstance.GetType().InvokeMember("Properties", BindingFlags.GetProperty, null, testContextInstance, null);

                    // Get a list of framework reserved properties so we can exclude them
                    foreach (var property in testContextInstance.GetType().GetProperties())
                    {
                        propeties.Add(property.Name);
                    }

                    foreach (KeyValuePair<string, object> property in contextProperties)
                    {
                        if (!propeties.Contains(property.Key) && property.Value is string)
                        {
                            // Add the override properties
                            passedInParameters.Add(property.Key, property.Value as string);
                        }
                    }
                }
                else if (NUnitTestContext.Parameters != null)
                {
                    // Update parameters for an NUnit unit test
                    TestParameters parameters = NUnitTestContext.Parameters;

                    foreach (string propertyName in parameters.Names)
                    {
                        // Add the override properties
                        passedInParameters.Add(propertyName, parameters[propertyName]);
                    }
                }

                // Update configuration values
                Config.AddTestSettingValues(passedInParameters);
            }
            catch (Exception e)
            {
                TryToLog(MessageType.WARNING, "Failed override configuration settings because: " + e.Message);
            }
        }
```

**Comment by TroyWalshProf** [comment]
This issue is fixed in the latest 5.5.0 release:
https://github.com/Magenic/MAQS/commit/2210264dfd52d2f0aaffb107cb92cd769516d670
